contrib/go-redis/redis/redis_test.go

```diff
@@ -313,3 +313,58 @@ func TestAnalyticsSettings(t *testing.T) {
 		assertRate(t, mt, 0.0, WithAnalyticsRate(0.0))
 	})
 }
+
+type testKey string
+
+func TestWithContextRace(t *testing.T) {
+	opts := &redis.Options{Addr: "127.0.0.1:6379"}
+	assert := assert.New(t)
+
+	k := testKey("key")
+	v := "val"
+	v2 := "val2"
+	ctx := context.WithValue(context.Background(), k, v)
+	client := NewClient(opts, WithServiceName("my-redis")).WithContext(ctx)
+	ctx2 := context.WithValue(context.Background(), k, v2)
+	client2 := client.WithContext(ctx2)
+
+	assert.Equal(ctx, client.Context())
+	assert.Equal(v, client.Context().Value(k))
+	assert.Equal(ctx2, client2.Context())
+	assert.Equal(v2, client2.Context().Value(k))
+}
+
+func TestWithContextProcess(t *testing.T) {
+	opts := &redis.Options{Addr: "127.0.0.1:6379"}
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	client := NewClient(opts, WithServiceName("my-redis"))
+	incorrectName := "incorrect.parent.span"
+	incorrectParent, ctx := tracer.StartSpanFromContext(context.Background(), incorrectName)
```

Comment:
These names are too long in my opinion. It takes too much to read the logic. I prefer to go by the reasoning that identifier names don't need to describe what the code is doing and that's something for documentation, if needed at all. Spending all this time reading identifier names and long strings obstructs the logic of what's going on. Just my 2c. 

I don't think we do this thoughout the rest of the repo: saving strings in variable names only to be checked later. Can't we just write the strings out everywhere instead and have variables like `root1`, `root2` or `span1`, `span2`.

* It can be misleading to call this `incorrectName`: what is incorrect about it?
* The same for `incorrectParent`: what if the test fails? Then it becomes super confusing that something called `incorrectParent` was actually correct. It almost feels like we're making assumptions about the outcome before we've asserted it.

Even if we were to describe the test in words, we would say: "_Create two spans and add them to the context of the client in order. Run an operation on the client and make sure that only the second span is part of the trace and becomes the parent of the operation span_" as opposed to "_Create one correct span and another incorrect span [...]_". As we are reading the latter without awareness of the context it becomes confusing about what is correct or incorrect about a span. When you write it with the entire context in mind it might make sense, but the reader doesn't have the context.

Reply:
>  The more descriptive names are longer, but I thought they helped clarify the code without documentation.

Let's please never use identifier names to clarify code, as a rule of thumb. Code is about logic, not words. Better keep the words for documentation. We should apply the naming rules from "[What's in a name](https://talks.golang.org/2014/names.slide#1)" everywhere.

If there is anything unclear or that you don't agree with about this philosophy we should talk about it and make sure we share the same mindset.

Reply:
👍